### PR TITLE
Support un-versioned `clang-tidy`

### LINKED ISSUE #392: `clang-tidy`/`clang-format` fallback
## What

Support the use of non-version-suffixed `clang-tidy` and `clang-format` as part of cmake `OPENASSETIO_ENABLE_CLANG_*` options. Document our minimum target clang version for code compliance.

## Why

The sufficed executable (symlink) seems to be a quirk of the `apt` installed LLVM on Ubuntu. Release distributions do not suffix these tools.

A side effect is that the executed version of the tool is uncontrolled and may result in variation between validations.

## ACs

- Update cmake to allow fallback to non-versioned tools
- Document policy of when using a newer version, satisfy its requirements and hope for backwards compatibility.

### PR DESCRIPTION
**Pull request by foundrytom**
Closes #392

### CODE REVIEW

File: cmake/StaticAnalyzers.cmake [hunk 1]
```diff
@@ -57,7 +58,8 @@ macro(enable_cpplint)
 endmacro()
 
 macro(enable_clang_format)
-    find_program(OPENASSETIO_CLANGFORMAT_EXE clang-format clang-format-10)
+    find_program(OPENASSETIO_CLANGFORMAT_EXE
+        NAMES clang-format clang-format-10)
```
Comment (themissingcow) [inline cmake/StaticAnalyzers.cmake:62]: @feltech Did we mean to have different versions for format vs tidy?

Comment (foundrytom) [reply cmake/StaticAnalyzers.cmake]: Capturing the off-line discussions, we decided to simply use un-versioned bin names for both `clang-tidy` and `clang-format` as described in the commit comment.

**Review by feltech — APPROVED** [review]
LGTM squash and merge when ready

File: cmake/StaticAnalyzers.cmake (hunk 1 above)
Comment (foundrytom) [reply cmake/StaticAnalyzers.cmake]: We re-added v12 support as some distros seem to only install with a suffix.............. 🙃

**Pull request merged**